src/webcam/react-webcam.tsx

```diff
@@ -0,0 +1,473 @@
+
+  static userMediaRequested = false
+
+  state = {
+    hasUserMedia: false,
+    mirrored: false,
+  }
+
+  stream?: MediaStream
+  canvas?: HTMLCanvasElement
+  ctx?: CanvasRenderingContext2D | null
+  video?: HTMLVideoElement | null
+  recordedBlobs?: Array<Blob>
+  mediaRecorder?: MediaRecorder
+
+  constructor(props: WebcamProps) {
+    super(props)
+
+    if (!hasGetUserMedia) {
+      const error = new Error('getUserMedia is not supported by this browser')
+      this.props.onFailure && this.props.onFailure(error)
+    }
+  }
+
+  async componentDidMount() {
+    Webcam.mountedInstances.push(this)
+    await this.requestUserMedia()
+    this.updateVideoElement()
+  }
+
+  componentDidUpdate() {
+    this.updateVideoElement()
+  }
+
+  updateVideoElement = () => {
+    const video = this.video
+    if (video) {
+      // The muted attributed is not currently being set due to a bug in react
+      // see: https://github.com/facebook/react/issues/10389
+      video.setAttribute('muted', 'true')
+    }
+  }
+
+  getConstraints(
+    width?: ConstrainULong,
+    height?: ConstrainULong,
+    facingMode?: FacingModeType,
+    audio?: boolean
+  ): MediaStreamConstraints {
+    /*
+    Safari 11 has a bug where if you specify both the height and width
+    constraints you must chose a resolution supported by the web cam. If an
+    unsupported resolution is used getUserMedia(constraints) will hit a
+    OverconstrainedError complaining that width is an invalid constraint.
+    This bug exists for ideal constraints as well as min and max.
+    However if only a height is specified safari will correctly chose the
+    nearest resolution supported by the web cam.
+    Reference: https://developer.mozilla.org/en-US/docs/Web/API/Media_Streams_API/Constraints
+    */
+
+    // if `{facingMode: 'user'}` Firefox will still allow the user to choose which camera to use (Front camera will be the first option)
+    // if `{facingMode: {exact: 'user'}}` Firefox won't give the user a choice and will show the front camera
+    const constraints: MediaStreamConstraints = {
+      video: { facingMode },
+      audio,
+    }
+
+    if (width) {
+      // @ts-ignore
+      constraints.video.width = parseInt(width, 10) || width // some devices need a Number type
+    }
+
+    if (height) {
+      // @ts-ignore
+      constraints.video.height = parseInt(height, 10) || height
+    }
+
+    return constraints
+  }
+
+  async requestUserMedia() {
+    if (!getUserMedia || !mediaDevices || Webcam.userMediaRequested) return
+    const {
+      width,
+      height,
+      facingMode,
+      audio,
+      fallbackWidth,
+      fallbackHeight,
+    } = this.props
+
+    const constraints = this.getConstraints(width, height, facingMode, audio)
+    const fallbackConstraints = this.getConstraints(
+      fallbackWidth,
+      fallbackHeight,
+      facingMode,
+      audio
+    )
+
+    const logError = (e: any) => console.log('error', e, typeof e)
```

Comment:
Since it's now part of the WebSDK we can use `trackException` and log errors to Sentry

Reply:
done